### PR TITLE
Finished analyses listing

### PR DESCRIPTION
**Pull request by gmrukwa**
* separated API and helpers
* helpers grouped by domain
* missing results endpoint added

### CODE REVIEW

**Comment by gmrukwa** [comment]
[DiviK results endpoint is not implemented yet](https://trello.com/c/hxNzNRIZ/423-divik-results-endpoint-is-not-implemented-yet)

File: test/discover/test_analysis.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    os.path.join(ROOT, 'data', 'peptides-2', 'divik', 'sample analysis'),
+    os.path.join(ROOT, 'data', 'peptides-2', 'divik', 'this is boring')
+]
+
+
+mock_only_existing = create_autospec(an.only_existing, side_effect=lambda f: f)
+mock_only_folders = create_autospec(
+    an.only_folders,
+    side_effect=lambda paths: [
+        path for path in paths if path in IS_DIR or path + '/' in IS_DIR
```
Comment (mg6) [inline test/discover/test_analysis.py]: `os.path.sep` needed here.

**Review by mg6 — CHANGES_REQUESTED** [review]

File: test/discover/test_analysis.py (hunk 1 above)
Comment (gmrukwa) [reply test/discover/test_analysis.py]: Accepted.

**Review by mg6 — DISMISSED** [review]

**Review by Daroslaw — DISMISSED** [review]

**Review by mg6 — APPROVED** [review]
Looks fine; good job.

**Review by Daroslaw — APPROVED** [review]

**Pull request merged**